### ISSUE TITLE
unecessary cast false positive

### ISSUE THREAD

**Issue by arifd** [rust-lang/rust-clippy#13878]
### Summary

This code will fail to compile:

```rs
fn main() {
    let n: f64 = 1.0;
    let _: f64 = (0.05).min(n);
}
```

making a cast necessary (or preferable to `0.05f64`).

Writing this code, will compile, but causes clippy to warn:
```rs
fn main() {
    let n: f64 = 1.0;
    let _: f64 = (0.05 as f64).min(n);
}
```

### Lint Name

unecessary cast

### Version

```text
rust playground stable 1.83.0: https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=115441b5376a81c65d1cd21c2d86ac6d
```

**Comment by Centri3** [comment]
> I suspect it might be really difficult to solve this in a non-hacky way

Seems doable :+1: Just need to check that the parent expression isn't one of the shared float methods (though trait methods would be fine here afaik.)

**Comment by arifd** [comment]
Yes, to be clear `0.05f64` will (correctly) silence the clippy warning, but I think `as f64` should also silence it.
